storage/range.go

```diff
@@ -179,6 +181,7 @@ func NewRange(desc *proto.RangeDescriptor, rm RangeManager) (*Range, error) {
 	r := &Range{
 		Desc:        desc,
 		rm:          rm,
+		firstIndex:  raftInitialLogIndex + 1,
```

Comment:
These are kind of weird because the first and last index are both inclusive instead of using a half-open interval. When the log is empty firstIndex must be lastIndex+1. Additionally, there is a dummy entry before the first real entry (used for term matching but nothing else). We could have initialized things as `firstIndex: raftInitialLogIndex, lastIndex: raftInitialLogIndex-1`, but this would have required more `-1`s sprinkled throughout the code in the handling of that first dummy entry. There are comments about this elsewhere but I'll add one here.